tests/vmi_multus_test.go

```diff
@@ -923,7 +923,6 @@ func checkMacAddress(vmi *v1.VirtualMachineInstance, interfaceName, macAddress s
 		&expect.BExp{R: "#"},
 		&expect.BSnd{S: cmdCheck},
 		&expect.BExp{R: macAddress},
-		&expect.BExp{R: "#"},
```

Comment:
I think the reason you removed it is because now there is no support for two sequential expecters.
It would have been more correct to fix the previous expect to:
```
&expect.BExp{R: macAddress + "((?s).*)" + PromptExpression},
```

I am fine with you just updating the commit message with the correct reasoning of this change (i.e. CheckForTextExpecter does not support two sequential expecters).

Reply:
I removed that expecter case because I saw error in logs about sequential expect cases.
`CheckForTextExpecter` appends `"((?s).*)" ` for each expect case for you
https://github.com/kubevirt/kubevirt/blob/ed4899ce43530db89b082d26d6b31835fe15842f/tests/utils.go#L2823
I will update the commit message

Reply:
> CheckForTextExpecter appends "((?s).*)" for each expect case for you

It appends it between what was previously sent and what you expect now (in this case, the mac address). It does not handle for you the processing of the buffer until the prompt (i.e. consuming the buffer after what you waited for until the prompt).
Because you leave that buffer, there is a potential wrong matching later. You should always strive to consume the buffer so this will not happen.